abseil.sh

```diff
@@ -5,33 +5,33 @@ requires:
   - "GCC-Toolchain:(?!osx)"
 build_requires:
   - CMake
+  - ninja
   - alibuild-recipe-tools
 source: https://github.com/abseil/abseil-cpp
+prefer_system: "osx"
+prefer_system_check: |
+  printf '#include <absl/container/flat_hash_map.h>' | c++ -std=c++17 -I`brew --prefix abseil`/include -c -xc++ - >/dev/null
 incremental_recipe: |
-  make ${JOBS:+-j$JOBS} install
+  cmake --build . -- ${JOBS:+-j$JOBS} install
   mkdir -p $INSTALLROOT/etc/modulefiles && rsync -a --delete etc/modulefiles/ $INSTALLROOT/etc/modulefiles
 ---
 #!/bin/bash -e
 
 mkdir -p $INSTALLROOT
 cmake $SOURCEDIR                             \
   ${CXXSTD:+-DCMAKE_CXX_STANDARD=$CXXSTD}    \
+  -DCMAKE_INSTALL_LIBDIR=lib                 \
   -DBUILD_TESTING=OFF                        \
   -DCMAKE_INSTALL_PREFIX=$INSTALLROOT
 
-make ${JOBS:+-j$JOBS} install
-
+cmake --build . -- ${JOBS:+-j$JOBS} install
 
 # Modulefile
 MODULEDIR="$INSTALLROOT/etc/modulefiles"
 MODULEFILE="$MODULEDIR/$PKGNAME"
 mkdir -p "$MODULEDIR"
-alibuild-generate-module > "$MODULEFILE"
+alibuild-generate-module --bin --lib > "$MODULEFILE"
 cat >> "$MODULEFILE" <<EoF
-
 # Our environment
-set ABSEIL_ROOT \$::env(BASEDIR)/$PKGNAME/\$version
-prepend-path PATH \$ABSEIL_ROOT/bin
-prepend-path LD_LIBRARY_PATH \$ABSEIL_ROOT/lib
-prepend-path LD_LIBRARY_PATH \$ABSEIL_ROOT/lib64
+prepend-path ROOT_INCLUDE_PATH \$PKG_ROOT/include
```

Comment:
this works for me with CVFMS

Reply:
so at least the productions on the GRID should be ok after this change

Reply:
actually, we need also to have o2 depend on abseil via `requires`, not a `build_requires`, otherwise nothing loads that module.

Reply:
You still need to have ROOT find the flat_map headers, which apparently rANS leaks to ROOT. A probably better solution would be to avoid having such implementation detail exposed to the headers seen by ROOT.

Reply:
I agree with @ktf : if we could hide the abseil headers, and make sure they are never parsed by ROOT, that would avoid such issues also in the future.

Reply:
Yes agreed. But we can do that in a second step, can't we? This PR is simple enough - is valid in any case - and brings back the productions.

Reply:
sure, if it works, LGTM. But let's not forget to clean it up and remove dependencies once we have a clean solution.

Reply:
Yes, if this works I guess we can survive with it, the problem is that so far it did not compile. I have dropped the ability to use the system abseil for now, since that was breaking macOS for whatever reason. Let's see. Worst case scenario we avoid breaking stuff with https://github.com/AliceO2Group/AliceO2/pull/11889 and then we try to reintroduce some flat_map.